app/src/main/java/com/example/packpals/viewmodels/TripsPageViewModel.kt

```diff
@@ -137,15 +137,18 @@ class TripsPageViewModel @Inject constructor(private val authRepo: AuthRepositor
                 if (creator==userId){
                     //reassign to a random trip pal if there are trip pals, else delete the trip
                     val tripPals = tripInfo.tripPalIds
-                    if(tripPals!=null){
-                        val randomPal = tripPals.shuffled()[0]
-                        //remove the random pal from the trip pals because now they're the creator
-                        tripPals?.toMutableList()?.remove(randomPal)
-                        val updatedTrip = Trip(trip.title, randomPal,tripPals,trip.active!!,trip.tripId)
-                        tripsRepo.updateTrip(trip.tripId!!, updatedTrip)
-                    }
-                    else{
-                        tripsRepo.deleteTrip(trip.tripId!!)
+                    if (tripPals != null) {
+                        if(tripPals.isNotEmpty()){
+                            val randomPal = tripPals.shuffled()[0]
+                            //remove the random pal from the trip pals because now they're the creator
+                            tripPals?.toMutableList()?.remove(randomPal)
```

Comment:
does this work? I thought toMutableList makes a copy so remove wouldn't actually remove from the original list

Reply:
no it doesn't LMFAO